packages/codec/lib/export.ts

```diff
@@ -272,3 +282,212 @@ function ethersCompatibleNativizeEventArgs(
+export class ReturndataDecodingInspector {
+  decoding: ReturndataDecoding;
+  constructor(decoding: ReturndataDecoding) {
+    this.decoding = decoding;
+  }
+  [util.inspect.custom](depth: number | null, options: InspectOptions): string {
+    switch (this.decoding.kind) {
+      case "return":
+        return formatFunctionLike(
+          "Returned values: ",
+          this.decoding.arguments,
+          options
+        );
+      case "returnmessage":
+        const { data } = this.decoding;
+        //we'll just set up a value and inspect that :)
+        const codecValue: Format.Values.BytesDynamicValue = {
+          kind: "value" as const,
+          type: {
+            typeClass: "bytes" as const,
+            kind: "dynamic" as const
+          },
+          value: {
+            asHex: data
+          }
+        };
+        const dataString = util.inspect(
+          new ResultInspector(codecValue),
+          options
+        );
+        return `Returned raw data: ${dataString}`;
+      case "selfdestruct":
+        return "The contract self-destructed.";
+      case "failure":
+        return "The transaction reverted without a message.";
+      case "revert":
+        const name = this.decoding.definedIn
+          ? `${this.decoding.definedIn.typeName}.${this.decoding.abi.name}`
+          : this.decoding.abi.name;
+        return formatFunctionLike(
+          `Error thrown:${OS.EOL}${name}`,
+          this.decoding.arguments,
+          options
+        );
+      case "bytecode":
+        //this one gets custom handling :P
+        const contractKind = this.decoding.class.contractKind || "contract";
+        const firstLine =
+          this.decoding.address !== undefined
+            ? `Returned bytecode for a ${contractKind} ${this.decoding.class.typeName} at ${this.decoding.address}.`
+            : `Returned bytecode for a ${contractKind} ${this.decoding.class.typeName}.`;
+        if (this.decoding.immutables && this.decoding.immutables.length > 0) {
+          const prefixes = this.decoding.immutables.map(
+            ({ name, class: { typeName } }) => `${typeName}.${name}: `
+          );
+          const maxLength = Math.max(...prefixes.map(prefix => prefix.length));
+          const paddedPrefixes = prefixes.map(prefix =>
+            prefix.padStart(maxLength)
+          );
+          const formattedValues = this.decoding.immutables.map(
+            (value, index) => {
+              const prefix = paddedPrefixes[index];
+              const formatted = indentExcludingFirstLine(
+                util.inspect(new ResultInspector(value.value), options),
+                maxLength
+              );
+              return prefix + formatted;
+            }
+          );
+          return `Immutable values:${OS.EOL}${formattedValues.join(OS.EOL)}`;
+        } else {
+          return firstLine;
+        }
+      case "unknownbytecode":
+        return "Bytecode was returned, but it could not be identified.";
+    }
```

Comment:
Is there any reason to include a `default` case in these? It could act as a check, but then again will TS catch these errors anyways?

Reply:
Well perhaps you could consider adding a `default` that throws?